src/main/java/com/exasol/adapter/dynamodb/mapping/JsonMappingFactory.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    private AbstractColumnMappingDefinition.ConstructorParameters readColumnProperties(final JsonObject definition,
+            final AbstractDynamodbResultWalkerBuilder resultWalker, final String dynamodbPropertyName)
+            throws MappingException {
+        final String destinationColumnName = readDestinationColumnName(definition, dynamodbPropertyName);
+        final AbstractColumnMappingDefinition.LookupFailBehaviour lookupFailBehaviour = readLookupFailBehaviour(
+                definition);
+        return new AbstractColumnMappingDefinition.ConstructorParameters(destinationColumnName, resultWalker.build(),
+                lookupFailBehaviour);
+    }
+
+    private ToStringColumnMappingDefinition.OverflowBehaviour readStringOverflowBehaviour(final JsonObject definition) {
+        if (definition.containsKey(OVERFLOW_KEY) && definition.getString(OVERFLOW_KEY).equals("ABORT")) {
+            return ToStringColumnMappingDefinition.OverflowBehaviour.EXCEPTION;
+        } else {
+            return DEFAULT_TO_STRING_OVERFLOW;
+        }
+    }
+
+    private AbstractColumnMappingDefinition.LookupFailBehaviour readLookupFailBehaviour(final JsonObject definition) {
+        if (definition.containsKey(REQUIRED_KEY) && definition.getBoolean(REQUIRED_KEY)) {
+            return AbstractColumnMappingDefinition.LookupFailBehaviour.EXCEPTION;
+        } else {
+            return DEFAULT_LOOKUP_BEHAVIOUR;
+        }
+    }
+
+    private String readDestinationColumnName(final JsonObject definition, final String defaultValue)
+            throws MappingException {
+        final String destinationColumnName = definition.getString(DEST_NAME_KEY, defaultValue);
+        if (destinationColumnName == null) {
+            throw new MappingException("Please set " + DEST_NAME_KEY + " property");
+        }
+        return destinationColumnName.toUpperCase();
+    }
+
+    private void addToJsonColumn(final JsonObject definition, final AbstractDynamodbResultWalkerBuilder resultWalker,
+            final TableMappingDefinition.Builder tableBuilder, final String dynamodbPropertyName)
+            throws MappingException {
+        final String destinationColumnName = readDestinationColumnName(definition, dynamodbPropertyName);
+        final AbstractColumnMappingDefinition.ConstructorParameters columnParameters = new AbstractColumnMappingDefinition.ConstructorParameters(
+                destinationColumnName, resultWalker.build(),
+                AbstractColumnMappingDefinition.LookupFailBehaviour.DEFAULT_VALUE);
+        tableBuilder.withColumnMappingDefinition(new ToJsonColumnMappingDefinition(columnParameters));
+    }
+
+    @Override
+    public SchemaMappingDefinition getSchemaMapping() {
+        return new SchemaMappingDefinition(this.tables);
+    }
+
+    /**
+     * Exception thrown if schema mapping definition is invalid.
+     */
+    public static class SchemaMappingException extends AdapterException {
```

Comment:
What's the difference between SchemaMappingException and MappingException? Do you need both of them?

Reply:
I think for users there is no difference why the exception happened. Good exception messages and hints on how to fix it help more than two different exception classes

Reply:
```
 /**
     * Exception that is thrown on mapping failures. This exceptions are caught and encapsulated into a
     * {@link SchemaMappingException} with additional info for the causing definition file.
     */
```
MappingException is caugh and extended by `causingMappingDefinitionFileName` in a SchemaMappingException

Reply:
At the place where `MappingExceptions` are thrown the file name is not accessible